Don't create multipart if not necessary

nkovacs opened the pull request:
Fixes #6

nkovacs committed:
Don't create multipart if not necessary

Add tests

Fixes #6

jpoehls commented on main.go:
```diff
@@ -177,50 +182,68 @@ func (m *Message) Bytes() ([]byte, error) {
 	}
 
 	// Does the message have a body?
 	if m.Body != "" || m.HTMLBody != "" {
```
I think we can remove this since we validate at the top that at least a `Body` or `HTMLBody` is non-empty.

nkovacs replied:
You're right, but actually I think it would be better to allow both to be empty, and just create a message with an empty plain text body and no html body.
This would also be backwards compatible.

jpoehls replied:
> You're right, but actually I think it would be better to allow both to be empty, and just create a message with an empty plain text body and no html body.

:+1:  I'm cool with that.

jpoehls commented:
Added a minor line note to check on but otherwise this looks good to me. Tests pass (we have tests finally!) -- Super excited to get this landed. Thanks for working on it!

nkovacs commented:
I've set up travis on my repo, and it's panicking on go master: https://travis-ci.org/nkovacs/gophermail/jobs/74920350

nkovacs committed:
Allow sending empty message.

If both bodies are empty, send an empty plain text body.

nkovacs committed:
Fix html+plain+attachment test under go 1.7

The test used the multipart reader incorrectly, which broke
under go 1.7.
When reading a part, the returned `multipart.Part` should be used
as the source, not the original buffer.
